### PR TITLE
OOP

### PR DESCRIPTION
**Pull request by TrOlyaDev**

### CODE REVIEW

File: src/oop/app/src/Parsers/FilmixParserStrategy.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+namespace src\oop\app\src\Parsers;
+
+class FilmixParserStrategy implements ParserInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function parseContent(string $siteContent)
+    {
+        $siteContent = iconv('CP1251',mb_detect_encoding($siteContent), $siteContent);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/FilmixParserStrategy.php]: missed space after 'CP1251',

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+use Symfony\Component\DomCrawler\Crawler;
+
+class KinoukrDomCrawlerParserAdapter implements ParserInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function parseContent(string $siteContent)
+    {
+        $crawler = new Crawler($siteContent);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: please use dependency injection for Crawler (inject it into constructor)

File: src/oop/app/src/Scrapper.php [hunk 3]
```diff
@@ -7,3 +7,36 @@
...
+namespace src\oop\app\src;
+
+use src\oop\app\src\Models\Movie;
+
+class Scrapper
+{
+    private $transporter;
+    private $parser;
+
+    public function __construct($transporter, $parser)
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Scrapper.php]: please use interfaces in constructor

File: src/oop/app/src/Scrapper.php [hunk 4]
```diff
@@ -7,3 +7,36 @@
...
+
+    /**
+     * Get film title, poster and description from third-party site
+     *
+     * @param string $url The site URL
+     * @return Movie Contains information about film title, poster and description
+     */
+    public function getMovie(string $url): Movie
+    {
+        $movie = new Movie();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Scrapper.php]: please use dependency injection for Movie (inject it into constructor)

File: src/oop/app/src/Transporters/GuzzleAdapter.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+class GuzzleAdapter implements TransportInterface
+{
+    /**
+     * @inheritDoc
+     * @throws \Exception
+     */
+    public function getContent(string $url): string
+    {
+        $client = new Client();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Transporters/GuzzleAdapter.php]: please use dependency injection for Client (inject it into constructor)

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/oop/app/src/Parsers/FilmixParserStrategy.php (hunk 1 above)
Comment (TrOlyaDev) [reply src/oop/app/src/Parsers/FilmixParserStrategy.php]: Fixed

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+use Symfony\Component\DomCrawler\Crawler;
+
+class KinoukrDomCrawlerParserAdapter implements ParserInterface
+{
+    private $crawler;
+
+    public function __construct()
+    {
+        $this->crawler = new Crawler();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: please do not use new keyword, use dependency injection:
public function __construct(Crawler $crawler)
{
   $this->crawler = $crawler;

File: src/oop/app/src/Scrapper.php [hunk 7]
```diff
@@ -7,3 +7,43 @@
...
+
+    /**
+     * @param TransportInterface $transporter
+     * @param ParserInterface $parser
+     */
+    public function __construct(TransportInterface $transporter, ParserInterface $parser)
+    {
+       $this->transporter = $transporter;
+       $this->parser = $parser;
+       $this->movie = new Movie();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Scrapper.php]: inject Movie same way as you did with parser

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

**Review by yderevianko-cartera — APPROVED** [review]

**Pull request merged**